File: leetcode-04/solutions/cpp/0661.cpp
class Solution {
 public:
  vector<vector<int>> imageSmoother(vector<vector<int>>& M) {
    const int m = M.size();
    const int n = M[0].size();

    vector<vector<int>> ans(m, vector<int>(n));

    for (int i = 0; i < m; i++)
      for (int j = 0; j < n; j++) ans[i][j] = helper(M, i, j);

    return ans;
  }

 private:
  int helper(vector<vector<int>>& M, int x, int y) {
    int sum = 0;
    int count = 0;

    for (int i = -1; i <= 1; i++)
      for (int j = -1; j <= 1; j++) {
        if (x + i < 0 || x + i >= M.size() || y + j < 0 || y + j >= M[0].size())
          continue;
        count++;
        sum += M[x + i][y + j];
      }

    return sum / count;
  }
};

File: leetcode-04/solutions/python3/0661.py
class Solution:
    def imageSmoother(self, M: List[List[int]]) -> List[List[int]]:
        m = len(M)
        n = len(M[0])

        ans = [[0 for j in range(n)] for i in range(m)]

        for i in range(m):
            for j in range(n):
                ans[i][j] = self.helper(M, i, j)

        return ans

    def helper(self, M: List[List[int]], x: int, y: int) -> int:
        sum = 0
        count = 0

        for i in range(-1, 2):
            for j in range(-1, 2):
                if x + i < 0 or x + i >= len(M) or \
                   y + j < 0 or y + j >= len(M[0]):
                    continue
                count += 1
                sum += M[x + i][y + j]

        return sum // count
